### DISCUSSION TITLE
how to use custom eslint plugin in CLIEngine class?

### DISCUSSION THREAD

**Discussion by xinqiymsz** [eslint/eslint#14133] [Q&A]
is the property of rulePaths? how to use rulePaths?what is the format?

**Comment by g-plane** [comment]
You just need to install ESLint plugins normally, then configure them in configuration file or override in constructor of `CLIEngine` class.

**Reply by xinqiymsz** [reply]
> You just need to install ESLint plugins normally, then configure them in configuration file or override in constructor of `CLIEngine` class.

“eslint-plugin-code-complexity” is my custom eslint plugin, can you show me code format?this is my code, but it will tip error. how can i write? thanks!
`const cli = new CLIEngine({
    parserOptions: {
        ecmaVersion: 2018,
        ecamFeatures: {
            jsx: true
        }
    },
    rules: {
         'eslint-plugin-code-complexity': ["error", { max: 20 }]
    },    
    useEslintrc: true
})`

**Reply by g-plane** [reply]
You may use `plugins` option in constructor of the exported `ESLint` class. [Documentation](https://eslint.org/docs/developer-guide/nodejs-api#-new-eslintoptions).

**Reply by xinqiymsz** [reply]
> You may use `plugins` option in constructor of the exported `ESLint` class. [Documentation](https://eslint.org/docs/developer-guide/nodejs-api#-new-eslintoptions).

i want to get eslint array message, so i use CLIEngine class
`const complexity = require("eslint-plugin-code-complexity")
const cli = new CLIEngine({
parserOptions: {}

rules: {
    eslint-plugin-code-complexity: ["error", { max: 20 }]
},
useEslintrc: true
})
const reports = cli.executeOnFiles(paths).results
`
i don't know how to introduce my plugin ,i has installed my eslint plugin “const complexity = require("eslint-plugin-code-complexity")”, how to deploy to CLIEngine class, use rules or rulepaths? what's the format

**Reply by g-plane** [reply]
We advise you to read documentation carefully.
